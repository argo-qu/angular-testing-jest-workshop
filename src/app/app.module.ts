import {NgModule} from '@angular/core';
import {BrowserModule} from '@angular/platform-browser';

import {AppRoutingModule} from './app-routing.module';
import {AppComponent} from './components/app-root/app.component';
import {SimpleService} from './services/simple/simple.service';
import {SimpleWithMocksService} from './services/simple-with-mocks/simple-with-mocks.service';

@NgModule({
  declarations: [
    AppComponent,
  ],
  imports: [
    BrowserModule,
    AppRoutingModule,
  ],
  providers: [
    SimpleService,
    SimpleWithMocksService
  ],
  bootstrap: [AppComponent],
})
export class AppModule {
}
