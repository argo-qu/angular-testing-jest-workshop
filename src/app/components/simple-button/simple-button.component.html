<button (click)="buttonClick.emit()">
  {{ text }}
</button>
