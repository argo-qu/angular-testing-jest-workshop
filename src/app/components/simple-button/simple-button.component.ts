import {Component, Input, Output, EventEmitter} from '@angular/core';

@Component({
  selector: 'app-simple-button',
  templateUrl: './simple-button.component.html',
  styleUrls: ['./simple-button.component.scss']
})
export class SimpleButtonComponent {

  @Input() text: string;

  @Output() buttonClick = new EventEmitter<any>();

}
