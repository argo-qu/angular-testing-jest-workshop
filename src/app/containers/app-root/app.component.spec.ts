import {TestBed} from '@angular/core/testing';
import {RouterTestingModule} from '@angular/router/testing';
import {AppComponent} from './app.component';
import {CUSTOM_ELEMENTS_SCHEMA} from '@angular/core';
import {SimpleButtonComponent} from '../../components/simple-button/simple-button.component';

describe('AppComponent', () => {

  // Юнит тесты
  describe('Unit', () => {
    let component: AppComponent;
    let fixture;

    beforeEach(async () => {
      await TestBed.configureTestingModule({
        schemas: [CUSTOM_ELEMENTS_SCHEMA],
        imports: [
          RouterTestingModule,
        ],
        declarations: [
          AppComponent,
        ],
      }).compileComponents();

      fixture = TestBed.createComponent(AppComponent);
      component = fixture.componentInstance;
    });

    it('should create the app', () => {
      expect(component).toBeTruthy();
    });

    describe('addNumbers', () => {
      it('should add numbers to the list', () => {
        expect(component.numbers).toEqual([]);
        component.addNumbers();
        expect(component.numbers).toEqual([100]);
        component.addNumbers();
        component.addNumbers();
        component.addNumbers();
        expect(component.numbers).toEqual([100, 200, 300, 400]);
      });
    });

    describe('layout', () => {
      it('should look properly', () => {
        // @ts-ignore
        expect(fixture.debugElement.nativeElement.innerHTML).toMatchSnapshot();

        component.addNumbers();
        fixture.detectChanges();

        // @ts-ignore
        expect(fixture.debugElement.nativeElement.innerHTML).toMatchSnapshot();

        component.addNumbers();
        component.addNumbers();
        component.addNumbers();
        fixture.detectChanges();

        // @ts-ignore
        expect(fixture.debugElement.nativeElement.innerHTML).toMatchSnapshot();
      });
    });

  });

  // Интеграционные тесты
  describe('Integration', () => {
    let component: AppComponent;
    let fixture;

    beforeEach(async () => {
      await TestBed.configureTestingModule({
        imports: [
          RouterTestingModule,
        ],
        declarations: [
          AppComponent,
          SimpleButtonComponent,
        ],
      }).compileComponents();

      fixture = TestBed.createComponent(AppComponent);
      component = fixture.componentInstance;
    });

    it('should create the app', () => {
      expect(component).toBeTruthy();
    });

    it('should add numbers when clicking button', () => {
      // @ts-ignore
      expect(fixture.debugElement.nativeElement.innerHTML).toMatchSnapshot();

      fixture.debugElement.nativeElement.querySelector('button').click();
      fixture.detectChanges();

      // @ts-ignore
      expect(fixture.debugElement.nativeElement.innerHTML).toMatchSnapshot();

      fixture.debugElement.nativeElement.querySelector('button').click();
      fixture.debugElement.nativeElement.querySelector('button').click();
      fixture.debugElement.nativeElement.querySelector('button').click();
      fixture.detectChanges();

      // @ts-ignore
      expect(fixture.debugElement.nativeElement.innerHTML).toMatchSnapshot();
    });
  });

});
