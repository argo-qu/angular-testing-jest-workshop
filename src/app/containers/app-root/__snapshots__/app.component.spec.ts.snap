// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppComponent Integration should add numbers when clicking button 1`] = `"<h1>Это просто кнопка</h1><app-simple-button text=\\"Add numbers\\"><button> </button></app-simple-button><ul><!----></ul>"`;

exports[`AppComponent Integration should add numbers when clicking button 2`] = `
"<h1>Это просто кнопка</h1><app-simple-button text=\\"Add numbers\\" ng-reflect-text=\\"Add numbers\\"><button> Add numbers
</button></app-simple-button><ul><!--bindings={
  \\"ng-reflect-ng-for-of\\": \\"100\\"
}--><li> 100 </li></ul>"
`;

exports[`AppComponent Integration should add numbers when clicking button 3`] = `
"<h1>Это просто кнопка</h1><app-simple-button text=\\"Add numbers\\" ng-reflect-text=\\"Add numbers\\"><button> Add numbers
</button></app-simple-button><ul><!--bindings={
  \\"ng-reflect-ng-for-of\\": \\"100\\"
}--><li> 100 </li><li> 200 </li><li> 300 </li><li> 400 </li></ul>"
`;

exports[`AppComponent Unit layout should look properly 1`] = `"<h1>Это просто кнопка</h1><app-simple-button text=\\"Add numbers\\"></app-simple-button><ul><!----></ul>"`;

exports[`AppComponent Unit layout should look properly 2`] = `
"<h1>Это просто кнопка</h1><app-simple-button text=\\"Add numbers\\"></app-simple-button><ul><!--bindings={
  \\"ng-reflect-ng-for-of\\": \\"100\\"
}--><li> 100 </li></ul>"
`;

exports[`AppComponent Unit layout should look properly 3`] = `
"<h1>Это просто кнопка</h1><app-simple-button text=\\"Add numbers\\"></app-simple-button><ul><!--bindings={
  \\"ng-reflect-ng-for-of\\": \\"100\\"
}--><li> 100 </li><li> 200 </li><li> 300 </li><li> 400 </li></ul>"
`;
